### PR TITLE
Resolved the managers issue - a get on BMC manager was giving 404 error

### PR DESCRIPTION
**Pull request by amar-shalgar**
1. Resolved the issue - A Get on a added dell server manager URL containing the ODIM generated UUID was returning 404 Not Found error instead of 200 OK
2. Even in the manager collection, the manager id for Dell server manager was not having UUID. Resolved this
3. The issue was if a system id and manager id is different then manager was giving 404, now handled this for any manager id

### CODE REVIEW

**Review by skhan-anjum — APPROVED** [review]

**Review by irappa-pattar — APPROVED** [review]

File: lib-utilities/common/mockconfig.go [hunk 1]
```diff
@@ -65,6 +65,12 @@ func SetUpMockConfig() error {
...
 	config.Data.AddComputeSkipResources = &config.AddComputeSkipResources{
 		SystemCollection: []string{
+			"Chassis",
+			"Managers",
+			"LogServices",
+		},
+		ManagerCollection: []string{
+			"Systems",
 			"Chassis",
 			"LogServices",
```
Comment (Bharath-KKB) [inline lib-utilities/common/mockconfig.go:75]: Are we sure LogServices is a manager?  What does Systems/ManagerCollection do ?

File: lib-utilities/config/config.go [hunk 2]
```diff
@@ -110,6 +110,7 @@ type APIGatewayConf struct {
 // AddComputeSkipResources stores all resource which need to igonered while adding Computer System
 type AddComputeSkipResources struct {
 	SystemCollection  []string `json:"SystemCollection"`  // holds the value of  system resource which need to be ignored
+	ManagerCollection []string `json:"ManagerCollection"` // holds the value of  manager resource which need to be ignored
```
Comment (Bharath-KKB) [inline lib-utilities/config/config.go]: why ignore these ? I think all these are badly named. Shouldnt they reflect the intended use ?

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
Answer queries and fix the issues.

File: lib-utilities/common/mockconfig.go (hunk 1 above)
Comment (amar-shalgar) [reply lib-utilities/common/mockconfig.go:75]: These collections are used to remove the links from the list of links present for that particular collection.
for e.g. - Managers links will be removed from list of links retrieved from System URI's and manager will be handled independently

Manager has LogServices links

File: lib-utilities/config/config.go (hunk 2 above)
Comment (amar-shalgar) [reply lib-utilities/config/config.go]: These collections are used for ignoring the links links from the list of all the links present for that URI

File: lib-utilities/common/mockconfig.go (hunk 1 above)
Comment (Bharath-KKB) [reply lib-utilities/common/mockconfig.go:75]: give these meaningful names.

File: lib-utilities/config/config.go (hunk 2 above)
Comment (Bharath-KKB) [reply lib-utilities/config/config.go]: Action the second comment to rename these

File: lib-utilities/config/odimra_config.json [hunk 3]
```diff
@@ -58,6 +58,13 @@
 	},
 	"AddComputeSkipResources": { 
 		"SystemCollection": [
```
Comment (Bharath-KKB) [inline lib-utilities/config/odimra_config.json]: Bad name

File: lib-utilities/config/odimra_config.json [hunk 4]
```diff
@@ -58,6 +58,13 @@
 	},
 	"AddComputeSkipResources": { 
 		"SystemCollection": [ 
+			"Chassis",
+			"Managers",
+			"LogServices",
```
Comment (Bharath-KKB) [inline lib-utilities/config/odimra_config.json:63]: I guess these URIs are present in the systems json . So what happens when a user queries for these URI ?

File: lib-utilities/config/odimra_config.json [hunk 5]
```diff
@@ -58,6 +58,13 @@
...
+			"WorkloadPerformanceAdvisor"
+		],
+		"ManagerCollection": [
```
Comment (Bharath-KKB) [inline lib-utilities/config/odimra_config.json]: Another bad name

File: odim-controller/helmcharts/odimra-config/templates/configmaps.yaml [hunk 6]
```diff
@@ -75,7 +75,14 @@ data:
     		"SystemCollection": [
     			"Chassis",
     			"LogServices",
-    			"WorkloadPerformanceAdvisor"
+                        "Managers",
```
Comment (Bharath-KKB) [inline odim-controller/helmcharts/odimra-config/templates/configmaps.yaml:78]: Why skip managers and Chassis under systems ?

File: odim-controller/helmcharts/odimra-config/templates/configmaps.yaml [hunk 7]
```diff
@@ -75,7 +75,14 @@ data:
...
+    			"WorkloadPerformanceAdvisor"                
+    		],
+            "ManagerCollection": [
+    			"Systems",
```
Comment (Bharath-KKB) [inline odim-controller/helmcharts/odimra-config/templates/configmaps.yaml:82]: why skip systems and chassis under managers ?

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
Address comments

File: lib-utilities/config/config.go (hunk 2 above)
Comment (Bharath-KKB) [reply lib-utilities/config/config.go]: Why was the comment marked as resolved when nothing was changed?

File: lib-utilities/config/odimra_config.json (hunk 3 above)
Comment (amar-shalgar) [reply lib-utilities/config/odimra_config.json]: Addressed comments - Modified the name

File: lib-utilities/config/odimra_config.json (hunk 5 above)
Comment (amar-shalgar) [reply lib-utilities/config/odimra_config.json]: Addressed comments - Modified the name

File: lib-utilities/config/config.go (hunk 2 above)
Comment (amar-shalgar) [reply lib-utilities/config/config.go]: Addressed comments - Modified the list names

File: lib-utilities/common/mockconfig.go (hunk 1 above)
Comment (amar-shalgar) [reply lib-utilities/common/mockconfig.go:75]: Addressed comments - Modified the list names

**Comment by amar-shalgar** [comment]
> Address comments

Addressed all the comments - modified the list names

File: odim-controller/helmcharts/odimra-config/templates/configmaps.yaml (hunk 7 above)
Comment (amar-shalgar) [reply odim-controller/helmcharts/odimra-config/templates/configmaps.yaml:82]: Chassis and Systems are already handled in a separate process, this list is added for ignoring the contents to be added in DB under manager

File: odim-controller/helmcharts/odimra-config/templates/configmaps.yaml (hunk 6 above)
Comment (amar-shalgar) [reply odim-controller/helmcharts/odimra-config/templates/configmaps.yaml:78]: Chassis and managers are already handled in a separate process, this list is added for ignoring the contents to be added in DB under system

File: lib-utilities/config/odimra_config.json (hunk 4 above)
Comment (amar-shalgar) [reply lib-utilities/config/odimra_config.json:63]: User will be able to access all the URI's. This list is for updating data into DB, while adding system details, manager,chassis and logservices are ignored as there is a separate process for it already

**Review by Bharath-KKB — APPROVED** [review]

**Pull request merged**